web3/contracts/FincubeDAO.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.8;
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract FincubeDAO is Ownable {
+    event MemberRegistered(address indexed _newMember, string _memberURI);
+    event ProposalCreated(
+        uint256 indexed proposalId,
+        address indexed newMember
+    );
+    event MemberApproved(address indexed member);
+    event ProposalExecuted(uint256 indexed proposalId);
+    uint256 private memberCount;
+
+    struct Member {
+        string memberURI;
+        bool status;
+    }
+    //@notice: This is own implementation of Proposal as per OpenZeppelin's governance function
+    struct Proposal {
+        address proposer;
+        uint48 voteStart;
+        uint48 voteDuration;
+        bool executed;
+        bool canceled;
+        address target;
+        bytes calldatas;
+        uint256 votes;
+    }
+
+    mapping(address => Member) private members;
+    mapping(uint256 => Proposal) private proposals;
+    uint256 private proposalCount;
+
+    constructor(string memory _ownerURI) Ownable(msg.sender) {
+        Member memory member;
+        member.memberURI = _ownerURI;
+        member.status = true;
+        members[msg.sender] = member;
+    }
+
+    function votingDelay() public pure returns (uint256) {
+        return 5; // 5 seconds
+    }
+
+    function votingPeriod() public pure returns (uint256) {
+        return 60; // 1 minute
+    }
+
+    function proposalThreshold() public view returns (uint256) {
+        return memberCount / 2;
+    }
+
+    modifier isNotExistingMember(address _address) {
+        require(
+            !(bytes(members[_address].memberURI).length > 0),
+            "Already a member"
+        );
+        _;
+    }
+
+    modifier onlyMember(address _address) {
+        require(
+            (bytes(members[_address].memberURI).length > 0),
+            "Not a member"
+        );
+        require(members[_address].status == true, "Member not approved");
+        _;
+    }
+
+    function register(
```

Comment:
For external and public functions please add @notice, @dev and @param NetSpec comments. Cz the @notice comment will help the general user to understand the basic idea of the business logic where the @dev and @param comments will be techincal so that Web2 and Web3 engineers can understand its actual implementation so that they can integrate the service.